test/lib/sealField.test.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+const assert = require('chai').assert;
+const mockery = require('mockery');
+const sinon = require('sinon');
+const symbolField = Symbol('symbol field');
+
+sinon.assert.expose(assert, { prefix: '' });
+
+describe('sealField', () => {
+    const password = 'aPasswordThatHasToBeAtLeast32CharactersLong';
+    const sealed = 'sealed value';
+    let ironMock;
+    let sealField;
+    let object;
+
+    before(() => {
+        mockery.enable({
+            useCleanCache: true,
+            warnOnUnregistered: false
+        });
+        ironMock = {
+            seal: sinon.stub(),
+            defaults: 'defaults'
+        };
+        mockery.registerMock('iron', ironMock);
+
+        // Lazy load dependency so that it registers the mock
+        // eslint-disable-next-line global-require
+        sealField = require('../../lib/sealField');
+    });
+
+    beforeEach(() => {
+        ironMock.seal.yieldsAsync(null, sealed);
+        object = {
+            field1: 'value 1',
+            field2: 'value 2',
+            [symbolField]: 'symbol value'
+        };
+    });
+
+    after(() => {
+        mockery.disable();
+    });
+
+    it('seals and updates the provided field', () =>
+        sealField(object, { field: 'field1', password })
+            .then(() => {
+                assert.calledWith(ironMock.seal, 'value 1', password, ironMock.defaults);
+                assert.strictEqual(object.field1, 'sealed value');
+            }));
+
+    it('returns the object', () =>
+        sealField(object, { field: 'field2', password })
+            .then((returned) => {
+                assert.strictEqual(returned, object);
+            }));
+
+    it('works on keys that are not strings', () =>
+        sealField(object, { field: symbolField, password })
+            .then(() => {
+                assert.calledWith(ironMock.seal, 'symbol value', password, ironMock.defaults);
+                assert.strictEqual(object[symbolField], 'sealed value');
+            }));
+
+    it('allows errors through', () => {
+        const err = new Error('an error');
+
+        ironMock.seal.yieldsAsync(err);
+
+        return sealField(object, { field: 'field1', password })
+            .then(() => {
```

Comment:
Could also be written as 

```
    .then(() => assert.fail('reason'), (e) => {
        // assert error case
    })
```

The way it's currently written, it will throw an error when the `assert.fail()` is reached. Then the `.catch` will have the "Should not get here" error instead of the true error that `sealField()` yields.